### ISSUE TITLE
consistent naming convention in API

### ISSUE THREAD

**Issue by ecoon** [hyriver/pynhd#82]
### Is your feature request related to a problem?

It would be nice to have a consistent API naming convention across the various data access classes.  Specifically:

https://github.com/hyriver/pynhd/blob/main/src/pynhd/pynhd.py#L727  --> `WaterData.byid()`

vs.

https://github.com/hyriver/pynhd/blob/main/src/pynhd/core.py#L224 --> `AGRBase.byids()`  (e.g. NHD, NHDPlusHR, etc)

I'm fairly sure that the semantics are the same (e.g. both accept a list of IDs, not just a single ID).

### Describe the solution you'd like

Change `WaterData.byid()` --> `WaterData.byids()` to match others.

**Comment by cheginit** [comment]
I agree with this. I've been planning to work on this for a while now, but haven't really had the chance to work on it. It's not just this name, there are other instances like `bygeom` vs `bygeometry`. This has to be done in backward compatible way, so essentially just new names are just aliases for the previous functions. A PR is appreciated.
